pyglui/ui_elements.pxi

```diff
@@ -929,25 +938,51 @@ cdef class Thumb(UI_element):
             utils.draw_circle(self.button.center,radius=int(min(self.button.size)*.77), stroke_width = stroke_width, color=RGBA(*thumb_color_shadow),sharpness=thumb_button_shadow_sharpness)
             utils.draw_circle(self.button.center,radius=int(min(self.button.size)*.77), stroke_width = stroke_width, color=self.off_color,sharpness=thumb_button_sharpness)
 
+        print 'button center: ', self.button.center
+        print 'radius: ', int(min(self.button.size)*.77)
+        print 'stroke_width: ', stroke_width
+
+
         if self.selected:
             self.selected = False
             global should_redraw
             should_redraw = True
 
         glfont.push_state()
+
+        # Choose either icon (fontawesome) or letter(roboto) label for button
+        cdef bytes text_to_draw
+        cdef float size_of_text
+        cdef list text_center_point
+
         glfont.set_font('roboto')
+        text_to_draw = self.label[0]
+        size_of_text = max(1,int(min(self.button.size))-thumb_font_padding)
+        text_center_point = [self.button.center[0],self.button.center[1]]
+
+        if self.use_icon_font:
+            glfont.set_font('fontawesome')
+            text_to_draw = self.label
+            size_of_text = 0.7*max(1,int(min(self.button.size))-thumb_font_padding)
+            text_center_point[0] = self.button.center[0] + self.icon_offset_x
+            text_center_point[1] = self.button.center[1] + self.icon_offset_y
+            if self.icon_scaling_num > 0:
+                size_of_text = self.icon_scaling_num*max(1,int(min(self.button.size))-thumb_font_padding)
+                
```

Comment:
Make `icon_scaling_num` default to `1.0`. Then you can get rid of the if-statement and line `966`.